streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -36,13 +42,17 @@
      * Create a new instance of {@link KStream} that consists of all elements of this stream which satisfy a predicate.
      *
      * @param predicate     the instance of {@link Predicate}
+     *
+     * @return a {@link KStream} containing only those records that satisfy the given predicate
```

Comment:
Perhaps "a KStream that contains only..." reads a bit better than "a KStream containing only..."